src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/RichTextEditorTextArea.kt

```diff
@@ -0,0 +1,65 @@
+package com.atlassian.performance.tools.jiraactions.api.page
+
+import com.atlassian.performance.tools.jiraactions.Patience
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.openqa.selenium.By
+import org.openqa.selenium.JavascriptExecutor
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.ExpectedConditions
+import org.openqa.selenium.support.ui.ExpectedConditions.*
+import java.time.Duration
+import org.openqa.selenium.support.ui.WebDriverWait
+
+
+
+/**
+ * Helper class for Rich Text Editor enabled Textarea
+ *
+ */
+class RichTextEditorTextArea(
+    private val driver: WebDriver
+) {
+
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+
+    fun overwriteIfPresent(id : String, text: String, optional : Boolean = false) {
+
+        //for optional field, ignore it if not found
+        if (optional && !driver.isElementPresent(By.id(id))) return
+
+        val wait = WebDriverWait(driver, 2)
+        val webElement = wait.until { driver -> driver.findElement(By.id(id)) }
+        val attrClass = webElement.getAttribute("class")
+        if(attrClass != null && attrClass.contains(other = "richeditor-cover", ignoreCase = true)){
+            overwriteRich(id, text)
+        }
+        else{
+            //try plain text editing
+            overwritePlain(webElement, text)
+        }
+    }
+
+    private fun overwritePlain(webElement: WebElement, text: String) {
+        driver.wait(timeout = Duration.ofSeconds(2), condition = elementToBeClickable(webElement))
+        webElement.clear()
+        webElement.sendKeys(text)
+    }
+
+    private fun overwriteRich(id: String, text: String) {
+        val iframeXpath = "//div[textarea[@id='$id']]//iframe"
```

Comment:
If we avoid re-finding the textarea and focus only on digging out the underlying iframe, then we can stop asking for `id: String` and expect an `textarea: WebElement` from the very top. That also means that `RichTextEditorTextArea` is no longer responsible for finding it and therefore doesn't have to handle optionality. If it's the `EditIssuePage`, which has the knowledge that the `description` textarea is optional, then let them make the call. Let them detect the presence and handle optionality.

Reply:
PS. Even better, we can then inject `textarea: WebElement` directly into the constructor.

Reply:
yep, that will be better, will do